File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

n = int(sys.stdin.readline())
stack = []

for i in range(n):
    command = list(map(int, sys.stdin.readline().split()))

    if command[0] == 1:
        stack.append(command[1])
    elif command[0] == 2:
        if stack:
            sys.stdout.write(f"{stack.pop()}\n")
        else:
            sys.stdout.write("-1\n")
    elif command[0] == 3:
        sys.stdout.write(f"{len(stack)}\n")
    elif command[0] == 4:
        if len(stack) == 0:
            sys.stdout.write("1\n")
        else:
            sys.stdout.write("0\n")
    elif command[0] == 5:
        if stack:
            sys.stdout.write(f"{stack[-1]}\n")
        else:
            sys.stdout.write("-1\n")

File: 백준/Silver/28278. 스택 2/스택 2.cc
#include <iostream>
#include <stack>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    int n;
    stack<int> s;
    cin >> n;

    for (int i = 0; i < n; i++) {
        int command;
        cin >> command;

        if (command == 1) {
            int x;
            cin >> x;
            s.push(x);
        }
        else if (command == 2) {
            if (!s.empty()) {
                cout << s.top() << "\n";
                s.pop();
            }
            else
                cout << -1 << "\n";
        }
        else if (command == 3)
            cout << s.size() << "\n";
        else if (command == 4) {
            if (s.empty())
                cout << 1 << "\n";
            else
                cout << 0 << "\n";
        }
        else if (command == 5) {
            if (!s.empty())
                cout << s.top() << "\n";
            else
                cout << -1 << "\n";
        }
    }
    
    return 0;
}